third_party/move/move-vm/runtime/src/instr.rs

```diff
@@ -0,0 +1,998 @@
+pub enum Instruction {
+    Pop,
+    Ret,
+    BrTrue(CodeOffset),
+    BrFalse(CodeOffset),
+    Branch(CodeOffset),
+    LdU8(u8),
+    LdU64(u64),
+    LdU128(Box<u128>),
+    CastU8,
+    CastU64,
+    CastU128,
+    LdConst(ConstantPoolIndex),
+    LdTrue,
+    LdFalse,
+    CopyLoc(LocalIndex),
+    MoveLoc(LocalIndex),
+    StLoc(LocalIndex),
+    Call(FunctionHandleIndex),
+    CallGeneric(FunctionInstantiationIndex),
+    Pack(StructDefinitionIndex),
+    PackGeneric(StructDefInstantiationIndex),
+    PackVariant(StructVariantHandleIndex),
+    PackVariantGeneric(StructVariantInstantiationIndex),
+    Unpack(StructDefinitionIndex),
+    UnpackGeneric(StructDefInstantiationIndex),
+    UnpackVariant(StructVariantHandleIndex),
+    UnpackVariantGeneric(StructVariantInstantiationIndex),
+    TestVariant(StructVariantHandleIndex),
+    TestVariantGeneric(StructVariantInstantiationIndex),
+    ReadRef,
+    WriteRef,
+    FreezeRef,
+    MutBorrowLoc(LocalIndex),
+    ImmBorrowLoc(LocalIndex),
+    MutBorrowField(FieldHandleIndex),
+    MutBorrowVariantField(VariantFieldHandleIndex),
+    MutBorrowFieldGeneric(FieldInstantiationIndex),
+    MutBorrowVariantFieldGeneric(VariantFieldInstantiationIndex),
+    ImmBorrowField(FieldHandleIndex),
+    ImmBorrowVariantField(VariantFieldHandleIndex),
+    ImmBorrowFieldGeneric(FieldInstantiationIndex),
+    ImmBorrowVariantFieldGeneric(VariantFieldInstantiationIndex),
+    MutBorrowGlobal(StructDefinitionIndex),
+    MutBorrowGlobalGeneric(StructDefInstantiationIndex),
+    ImmBorrowGlobal(StructDefinitionIndex),
+    ImmBorrowGlobalGeneric(StructDefInstantiationIndex),
+    Add,
+    Sub,
+    Mul,
+    Mod,
+    Div,
+    BitOr,
+    BitAnd,
+    Xor,
+    Or,
+    And,
+    Not,
+    Eq,
+    Neq,
+    Lt,
+    Gt,
+    Le,
+    Ge,
+    Abort,
+    Nop,
+    Exists(StructDefinitionIndex),
+    ExistsGeneric(StructDefInstantiationIndex),
+    MoveFrom(StructDefinitionIndex),
+    MoveFromGeneric(StructDefInstantiationIndex),
+    MoveTo(StructDefinitionIndex),
+    MoveToGeneric(StructDefInstantiationIndex),
+    Shl,
+    Shr,
+    VecPack(SignatureIndex, u64),
+    VecLen(SignatureIndex),
+    VecImmBorrow(SignatureIndex),
+    VecMutBorrow(SignatureIndex),
+    VecPushBack(SignatureIndex),
+    VecPopBack(SignatureIndex),
+    VecUnpack(SignatureIndex, u64),
+    VecSwap(SignatureIndex),
+    PackClosure(FunctionHandleIndex, ClosureMask),
+    PackClosureGeneric(FunctionInstantiationIndex, ClosureMask),
+    CallClosure(SignatureIndex),
+    LdU16(u16),
+    LdU32(u32),
+    LdU256(Box<U256>),
+    CastU16,
+    CastU32,
+    CastU256,
+    LdI8(i8),
+    LdI16(i16),
+    LdI32(i32),
+    LdI64(i64),
+    LdI128(Box<i128>),
+    LdI256(Box<I256>),
+    CastI8,
+    CastI16,
+    CastI32,
+    CastI64,
+    CastI128,
+    CastI256,
+    Negate,
+
+    VecLenV2,
+    VecSwapV2,
+    TestVariantV2(TestVariantV2),
+    BorrowFieldV2(Box<BorrowFieldV2>),
```

Comment:
Why box? Our instruction large enough, we do not need to have any indirection, especially on execution hot path (eg without accessing types), having box here seems very weird?

Having index into side table is also better - then we have cache locality for instructions which are nearby (e.g. borrow_field then test_variant)